# Defensive Investment Strategy
Investing into Indexes is a lower risk investment strategy than indivually picking out stocks. The benchmark to beat market returns is hard for even professional investors and many fund managers make worse than average returns. 

As a defensive investor, instead of speculating the market and trying to pick out invidual stocks, its better to invest into an index fund and get the average market return. However, instead of setting up a regular payment to invest into a fund, what if we could take advantage of trading an index fund? Or better yet, if we could also take advantage of daily price movements of indexes.

The index ETFs might be one way to do this, as they allow investors to place limit orders on major indexes such as FTSE 100, S&P 500 etc. We need a strategy to take advantage of price swings on indexs.


### Quest to find an optimum strategy to invest £25,000 a year into index tracking ETFs

Stocks and Shares ISA account allows £20,000 and 
Lifetime ISA allows £4,000 (+£1,000 government bonus) to be invested annually
tax free. This gives us a total of £25,000 to be invested tax free.

The goal is to use **pound cost averaging** strategy alongside another strategy to
get the same or higher returns than the FTSE 100 average return. If these strategies work,
then they can be extended to other indexes like S&P 500 etc. to diversify the portfolio.

The idea is to invest 10-20 times a year, at the right time, in index tracking
ETF's, as well as take advantage of daily price swings, then we can at the **minimum** keep 
up with the average market return.


**Benchmark:**
Our benchmark is investing into index funds once a month on a random day e.g. 
28th of every month in FTSE 100.


#### The main challenge
The hard part of this strategy is to beat the benchmark despite the higher dealing charges and fees of trading ETFs. Low cost index fund for FTSE 100 have an ongoing annual charge of 0.06% with £0 cost to buy or sell. On the other hand, investing into ETF's cost £10 (AJBell) to buy and £10 to sell, as well as an ongoing annual charge of 
0.07%. This means that our trading strategy needs to give us atleast slightly better return to cover the dealing cost,
otherwise, we might do slightly worse than the market.


#### Summary

- Invest £1,250 to £2,500 per trade.
- Invest 10-20 throughout the year (pound cost averaging).
- Predict the price movement of FTSE 100 continously and invest once every 20-60 day period.
- Come up with a strategy to take advantage of daily price movements.
- Find a strategy to sell (not urgent as we're investing for mid/long term).



## FTSE 100 - EDA

In [4]:
import pandas as pd

In [5]:
df = pd.read_csv('../data/raw/download.csv')

### Cleaning up data

In [6]:
sheet = '{}{}{}'.format('https://docs.google.com/spreadsheets/d/',
                        '10cihRP6XxJOSrWsASfW5C7VeHufD32TS0JKCXLoWhL8',
                        '/export?gid=0&format=csv')

In [41]:
df = pd.read_csv(sheet)

In [42]:
df.head(2)

Date  Unnamed: 1    High     Low   Close  Volume
0  02/01/1990 16:30:00      2442.4  2443.3  2425.5  2434.1       0
1  03/01/1990 16:30:00      2451.3  2466.2  2445.8  2463.7       0

In [90]:
# Rename columns
df = df.rename(columns={'Unnamed: 1': 'Open'})

# Convert to datetime
df['Date'] = pd.to_datetime(df['Date'])

# Get datetime related fields
df['month'] = df['Date'].dt.month_name()
df['year'] = df['Date'].dt.year
df['dat_name'] = df['Date'].dt.day_name()
df['weekofyear'] = df['Date'].dt.weekofyear
df['is_quarter_end'] = df['Date'].dt.is_quarter_end
df['is_quarter_start'] = df['Date'].dt.is_quarter_start
df['quarter'] = df['Date'].dt.quarter

In [91]:
df.tail(2)

Date     Open     High      Low    Close  Volume  \
7626 2020-03-18 16:30:00  5294.90  5294.90  5006.20  5080.58       0   
7627 2020-03-19 16:30:00  5080.58  5181.04  4942.37  5151.61       0   

      high_low_diff  open_close_diff  month  year   dat_name  weekofyear  \
7626           5.77            -4.05  March  2020  Wednesday          12   
7627           4.83             1.40  March  2020   Thursday          12   

      is_quarter_end  is_quarter_start  quarter  
7626           False             False        1  
7627           False             False        1

### Days with the highest difference between high and low price (%)

In [55]:
df['high_low_diff'] = round(((df['High'] - df['Low'])/df['Low']) * 100, 2)
df.sort_values(by='high_low_diff', ascending=False).head(5)

Date     Open     High      Low    Close  Volume  \
7622 2020-12-03 16:30:00  5876.52  5876.52  5237.48  5237.48       0   
4738 2008-10-10 16:30:00  4313.80  4313.80  3873.99  3932.06       0   
4723 2008-09-19 16:30:00  4880.00  5351.20  4857.10  5311.30       0   
4748 2008-10-24 16:30:00  4087.83  4087.83  3715.24  3883.36       0   
4769 2008-11-24 16:30:00  3780.96  4153.08  3780.96  4152.96       0   

      high_low_diff  open_close_diff  
7622          12.20           -10.87  
4738          11.35            -8.85  
4723          10.17             8.84  
4748          10.03            -5.00  
4769           9.84             9.84

In [56]:
p = 'Mean difference between daily high and low over the last 20 year: {}%'.format(df['high_low_diff'].mean().round(2))
print(p)

Mean difference between daily high and low over the last 20 year: 1.28%


### High/Low difference by month

In [96]:
agg_df = (
    df[['year', 'month', 'High', 'Low']]
    .groupby(['year', 'month'])
    .agg({'High': 'max', 'Low': 'min'})
    .reset_index()
    )

agg_df['high_low_diff'] = round(((agg_df['High'] - agg_df['Low'])/agg_df['Low']) * 100, 2)

In [100]:
agg_df.sort_values(by='high_low_diff', ascending=False).head(10)

year     month     High      Low  high_low_diff
226  2008   October  6314.50  3665.21          72.28
225  2008  November  6251.80  3734.07          67.43
219  2008  February  6512.30  3973.26          63.90
223  2008     March  6487.80  4042.31          60.50
216  2008     April  6534.70  4090.04          59.77
224  2008       May  6377.00  4002.21          59.34
367  2020     March  7626.41  4898.79          55.68
220  2008   January  6247.30  4038.45          54.70
235  2009     March  5372.18  3497.27          53.61
228  2009     April  5373.21  3512.09          52.99

#### FTSE 100 in October 2008

We see that during October 2008, the highest price was 6314 and the lowest 3665, which a percentage difference of a massive 72%!!!

#### Dates of highest high and lowest low in October 2008

In [125]:
(
    df
    .loc[(df['year'] == 2008) & (df['month'] == 'October'), ['High', 'Date']]
    .sort_values(by='High', ascending=False)
    .head(5)
)

High                Date
4547  6314.5 2008-10-01 16:30:00
4610  6003.2 2008-10-04 16:30:00
4651  5877.6 2008-10-06 16:30:00
4589  5718.8 2008-10-03 16:30:00
4673  5529.6 2008-10-07 16:30:00

In [126]:
(
    df
    .loc[(df['year'] == 2008) & (df['month'] == 'October'), ['Low', 'Date']]
    .sort_values(by='Low', ascending=True)
    .head(5)
)

Low                Date
4749  3665.21 2008-10-27 16:30:00
4748  3715.24 2008-10-24 16:30:00
4750  3847.55 2008-10-28 16:30:00
4742  3861.39 2008-10-16 16:30:00
4738  3873.99 2008-10-10 16:30:00

We see that the highest price in October 2008 was £6,314.50 on the 1st and the lowest price was £3,665.21 on the 27th.

## Strategy - 10 day Average lows

There is no way to predict that the price would've dropped down to 3665 at the end of the month. However, what if we took the average lows for the first 10 days of October and put a limit order of anything below that average price. Would we have gotten a better price? 

Lets find out!

### Days with the highest difference between Open and Close Price (%)

In [48]:
df['open_close_diff'] = round(((df['Close'] - df['Open'])/df['Open']) * 100, 2)

#### Biggest decrease between open and close

In [25]:
df.sort_values(by='open_close_diff', ascending=True).head(5)

Open     High      Low    Close  Volume  \
Date                                                              
12/03/2020 16:30:00  5876.52  5876.52  5237.48  5237.48       0   
10/10/2008 16:30:00  4313.80  4313.80  3873.99  3932.06       0   
06/10/2008 16:30:00  4980.25  4980.25  4549.66  4589.19       0   
09/03/2020 16:30:00  6462.55  6462.55  5891.56  5965.77       0   
11/09/2001 16:30:00  5033.70  5129.00  4746.00  4746.00       0   

                     high_low_diff  open_close_diff  
Date                                                 
12/03/2020 16:30:00          12.20           -10.87  
10/10/2008 16:30:00          11.35            -8.85  
06/10/2008 16:30:00           9.46            -7.85  
09/03/2020 16:30:00           9.69            -7.69  
11/09/2001 16:30:00           8.07            -5.72

#### Biggest increase between open and close

In [23]:
df.sort_values(by='open_close_diff', ascending=True).tail(5)

Open     High      Low    Close  Volume  \
Date                                                              
13/03/2003 16:30:00  3287.00  3489.10  3287.00  3486.90       0   
29/10/2008 16:30:00  3926.38  4242.54  3926.38  4242.54       0   
13/10/2008 16:30:00  3932.06  4256.90  3932.06  4256.90       0   
19/09/2008 16:30:00  4880.00  5351.20  4857.10  5311.30       0   
24/11/2008 16:30:00  3780.96  4153.08  3780.96  4152.96       0   

                     high_low_diff  open_close_diff  
Date                                                 
13/03/2003 16:30:00           6.15             6.08  
29/10/2008 16:30:00           8.05             8.05  
13/10/2008 16:30:00           8.26             8.26  
19/09/2008 16:30:00          10.17             8.84  
24/11/2008 16:30:00           9.84             9.84

In [37]:
25000/10

2500.0

In [28]:
24000/12

2000.0